web-components/src/components/form/Form.ts

```diff
@@ -0,0 +1,189 @@
+    @property({ type: Boolean, reflect: true, attribute: "is-valid" }) isvalid = false;
+
+    @query(".md-form") protected form!: HTMLFormElement;
+
+    private formElement: HTMLFormElement[] = [];
+
+    private isSubmittable(element: HTMLElement) {
+      if (
+        element instanceof HTMLInputElement ||
+        element instanceof HTMLSelectElement ||
+        element instanceof HTMLTextAreaElement
+      ) {
+        return true;
+      }
+      return false;
+    }
+
+    private isDisabled(element: HTMLElement) {
+      return element.hasAttribute("disabled") || element.getAttribute("aria-disabled") === "true";
+    }
+
+    private findSubmittable(
+      parent: HTMLElement | ShadowRoot,
+      submittable: Set<SubmittableElement>
+    ): SubmittableElement[] {
+      const children = Array.from(parent.children) as HTMLElement[];
+
+      for (const child of children) {
+        if (this.isDisabled(child)) {
+          continue;
+        }
+
+        if (this.isSubmittable(child)) {
+          submittable.add(child as SubmittableElement);
+        }
+
+        if (child.shadowRoot) {
+          this.findSubmittable(child.shadowRoot, submittable);
+        } else if (child.tagName === "SLOT") {
+          const childElements = (child as HTMLSlotElement).assignedElements();
+          if (childElements.length) {
+            const { parentElement } = childElements[0];
+            if (parentElement) {
+              this.findSubmittable(parentElement, submittable);
+            }
+          }
+        } else {
+          this.findSubmittable(child, submittable);
+        }
+      }
+      return [...submittable];
+    }
+
+    private validated() {
+      if (this.novalidate) {
+        return true;
+      }
+      return this.isvalid;
+    }
+
+    handleSubmit = (event: Event) => {
+      event.preventDefault();
+
+      if (this.validated()) {
+        this.submitForm();
+      }
+    };
+
+    private submitForm() {
+      const { form } = this;
+
+      if (form.requestSubmit) {
+        form.requestSubmit();
+      } else {
+        form.submit();
+      }
+    }
+
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      this.cleanupWrappedForms();
+    }
+
+    private cleanupWrappedForms() {
+      const { formElement } = this;
+      if (formElement.length) {
+        for (const element of formElement) {
+          this.deleteWrappedForm(element);
+        }
+      }
+
+      this.formElement = [];
+    }
+
+    private deleteWrappedForm(element: HTMLFormElement) {
+      element.removeEventListener("submit", this.handleSubmit);
+    }
+
+    private createWrappedForm(element: SubmittableElement) {
+      const form = document.createElement("form");
+
+      form.addEventListener("submit", this.handleSubmit);
+
+      const { parentNode: parent } = element;
+
+      if (!element.id) {
+        element.id = "input-form";
+      }
+      if (!element.name) {
+        element.name = "input-form";
```

Comment:
this actually needs to be configurable. only when i change this value to "name" in the DOM manually, I get the autofill window: 
<img width="1278" alt="Screen Shot 2021-01-26 at 12 53 31 PM" src="https://user-images.githubusercontent.com/3697148/105904468-05cea180-5fd6-11eb-8eb9-fe58fac3b62f.png">

Reply:
So i added this myself (name attribute to form). Please take notes next time we are discussing action items.